examples/distributed/pyg/dist_train_sage_for_homo.py

```diff
@@ -0,0 +1,348 @@
+import argparse
+import os.path as osp
+import time
+
+import torch
+import torch.distributed
+import torch.nn.functional as F
+from ogb.nodeproppred import Evaluator
+from torch.nn.parallel import DistributedDataParallel
+from tqdm import tqdm
+
+import torch_geometric.distributed as pyg_dist
+from torch_geometric.distributed import LocalFeatureStore, LocalGraphStore
+from torch_geometric.distributed.dist_context import DistContext, DistRole
+from torch_geometric.distributed.partition import load_partition_info
+from torch_geometric.nn import GraphSAGE
+from torch_geometric.typing import Tuple
+
+
+@torch.no_grad()
+def test(model, test_loader, dataset_name):
+    evaluator = Evaluator(name=dataset_name)
+    model.eval()
+    xs = []
+    y_true = []
+    for i, batch in enumerate(test_loader):
+        if i == 0:
+            device = batch.x.device
+        x = model(batch.x, batch.edge_index)[:batch.batch_size]
+        xs.append(x.cpu())
+        y_true.append(batch.y[:batch.batch_size].cpu())
+        del batch
+        if i == len(test_loader) - 1:
```

Comment:
I suggest that we remove this barrier wait here and instead in the enable `persistent_workers=True` in the `DistNeighborLoader` when `num_workers > 0`. This should allow for the RPC to stay alive in worker subprocesses and allow for processing the remote requests if one of the iterator exits sooner than the other (when async processing is on).
I tested this on my system, didn't get any RPC connection issues.
Would you please add such a line to the `dist_neighbor_loader.py`?
`NodeLoader.__init__(
            self,
             ...
            persistent_workers=True if kwargs['num_workers'] > 0 else False,
            )`

Reply:
The connectivity issue should be fixed by https://github.com/pyg-team/pytorch_geometric/pull/8605, no need to use `persistent_workers`